accessibility/README.md

```diff
@@ -0,0 +1,490 @@
+</button>
+
+<a href="/account">
+  <icon-account />
+</a>
+
+<a href="/sale">
+  <!-- Image containing embedded text about a 10% off sale -->
+  <img src="10-off-promo-image.jpg">
+</a>
+
+
+```
+
+#### Do
+
+```vue
+<a class="site-logo" href="/">
+  <brand-logo />
+  
+  <span class="visually-hidden">
+    My Shop Name
+  </span>
+</a>
+
+<button
+  type="button"
+  @click="openMenuDrawer"
+>
+  <icon-menu />
+  
+  <span class="visually-hidden">
+    Toggle Menu
+  </span>
+</button>
+
+<a href="/account">
+  <icon-account />
+  
+  <span class="visually-hidden">
+    Account
+  </span>
+</a>
+
+<a href="/sale">
+  <!-- Image containing embedded text about a 10% off sale -->
+  <img src="10-off-promo-image.jpg" alt="Get 10% off">
+</a>
+```
+
+* When using `<button>` or `<a>` elements containing SVG elements, ensure that accessible text is also included
+  * If the text should not be visible, the `visually-hidden` class can be used
+* If a `<button>` or `<a>` contains a `<img>` element - ensure that there is either appropriate `alt` text on the image, or add a supporting `visually-hidden` span with appropriate text
+
+### Ensure correct usage of `aria-labelledby` and `aria-label` 
+
+* The `aria-labelledby` and `aria-label` attributes provide another method for adding labels to elements
+* These are alternatives to using a `visually-hidden` element showcased in the examples above
+
+#### `aria-labelledby`
+
+* `aria-labelledby` can be used to label an element based on the text content of another existing element by it's unique `id`
+* If a heading is present inside a region, this could be a good candidate for `aria-labelledby`
+
+```html
+<section aria-labelledby="template-title">
+  <h1 id="template-title">
+    Title for the section
+  </h1>
+
+  <!-- other content -->
+</section>
+```
+* Ensure that the element referenced by `aria-labelledby` exists on the page.
+  * If the referenced element is missing, screen readers will still read out the the other text content contained in the element, but users may be missing out on the additional context that the referenced element might be adding
+* When using `aria-labelledby` in a component, consider whether the element you are referencing existing inside the component's template code.
+  * If the element you are targeting with `aria-labelledby` is stored in another component or snippet, you may not be able to guarantee that it will always be on the page.
+  * In this case, opt to use `<span class="visually-hidden">` or `aria-label` instead.
+* Additional documentation: [MDN - aria-labelledby](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-labelledby)
+
+#### `aria-label`
+* `aria-label` is another approach for labelling elements, and can be used when there is no visible element in the DOM that can be used directly as a label or referenced with `aria-labelledby`
+* Examples of using this are similar to adding a `<span class="visually-hidden">` element
```

Comment:
Would it be worth adding some context why we're using `visually-hidden` text in the above examples?

I think I generally use them because `aria-label` has had a rocky history with screenreader support ([see this blog post from 2020](https://gomakethings.com/revisting-aria-label-versus-a-visually-hidden-class/)), and printed text is always read out (unless `aria-hidden` or `display: none` is present - but I don't know if that's actually the reason.

Reply:
This is good to know - I've always preferred using `visually-hidden` over `aria-label` - so its good to have some reasons why to back this up.
I've added some notes based on that blog post and linked to it directly

Reply:
Oooh I didn't even think about translation! Nice!